fb.c

```diff
@@ -145,3 +145,148 @@ void destroy_framebuffer(const char *proc_fname)
     pr_debug("Destroying framebuffer %s\n", proc_fname);
     remove_proc_entry(proc_fname, NULL);
 }
+
+static int vcam_fb_check_var(struct fb_var_screeninfo *var,
+                             struct fb_info *info)
+{
+    if (!var->xres)
+        var->xres = 1;
+    if (!var->yres)
+        var->yres = 1;
+    if (var->xres > var->xres_virtual)
+        var->xres_virtual = var->xres;
+    if (var->yres > var->yres_virtual)
+        var->yres_virtual = var->yres;
+
+    var->xres_virtual = var->xoffset + var->xres;
+    var->yres_virtual = var->yoffset + var->yres;
+
+    var->bits_per_pixel = 24;
+
+    switch (var->bits_per_pixel) {
+    case 24:
```

Comment:
Only bpp = 24 is supported?

Reply:
It is the screeninfo variable, so It can support rgb type for example rgb565, rgb5551.